roles/dlrn_report/tasks/dlrn_report_results.yml

```diff
@@ -5,6 +5,12 @@
     name: set-zuul-log-path-fact
 
 - name: Get hash related data from repo_setup role
+  vars:
+    cifmw_repo_setup_promotion: >-
+      {{
+        (cifmw_dlrn_report_promote_source | length > 0) |
+        ternary(cifmw_dlrn_report_promote_source, cifmw_repo_setup_promotion)
+      }}
```

Comment:
@pablintino Hello, I am not sure why I am hitting [following error](https://logserver.rdoproject.org/70/2170/3e33d800f341ce9c10a996876730075edbf2b1b4/github-check/cifmw-pod-pre-commit/c845bd8/pre_commit.log):
```
Error was a <class 'ansible.errors.AnsibleError'>, original message: recursive loop detected in template string: {{
  (cifmw_dlrn_report_promote_source | length > 0) |
  ternary(cifmw_dlrn_report_promote_source, cifmw_repo_setup_promotion)
}}. maximum recursion depth exceeded while calling a Python object
```

Reply:
Thank you @ciecierski , I have changed to simpler set_fact. The task will also look much simpler now.